helion/_compiler/compile_environment.py

```diff
@@ -99,6 +99,14 @@ def allocate_block_size(
         hint: int = 64,
     ) -> int:
         idx = len(self.block_sizes)
+        from .host_function import HostFunction
+
+        if isinstance(size, torch.SymInt):
+            # If block size is already allocated, we reuse the existing index.
```

Comment:
Where is this getting called from?  I'd expect this to get caught earlier, and the caller will expect a new block size.

Reply:
For the matmul_layernorm example:
```python
    n = hl.register_reduction_dim(y.size(1))
    for tile_m in hl.tile(m):
        for tile_k in hl.tile(k):
            mm = torch.matmul(x[tile_m, tile_k], y[tile_k, :])
```
It's for the `:` in `y[tile_k, :]` to be able to look up the reduction dim registered by `hl.register_reduction_dim(y.size(1))`.

Agreed that `allocate_block_size()` should be strictly for allocating a new block size - I moved this logic to beginning of `allocate_reduction_dimension()` instead, where we already have similar logic for checking for existing reduction dimensions.